### DISCUSSION TITLE
External clock and D-flip-flop (for a GAL16V8)?

### DISCUSSION THREAD

**Discussion by rehsd** [hneemann/Digital#1175] [Q&A] [answered]
I am building some basic logic for use on a GAL16V8. I would like to bring an external clock into the GAL and latch a pair of outputs using a D-flip-flop. I tried to use an Input on pin 1 and connecting that to the clock on the flip-flops. When clicking on Analysis\Analysis, I get an error "A single clock component is necessary. All flip-flops must use this clock signal." I then swapped out the Input with Clock Input. When I program the GAL, it seems like the Clock Input is behaving as an output and affecting the system clcok (external to the GAL). I am likely missing something simple. How do I use an external clock as an input to drive D-flip-flops? Thank you!  --Rich

![ext_clock](https://github.com/hneemann/Digital/assets/87081882/4f025dc9-f554-434b-963b-b7ed1d66a030)

**Comment by hneemann** [comment, marked as answer]
I'm not sure what's happening here. When the clock symbol is used and the JDEC file is created, the clock input on pin 1 is used. Pin 1 cannot be used as an output either. Pin 1 is physically only an input. The GAL16v8 has also no other possibility for a clock input. Could it be an interpretation problem and everything is working correctly? Or the problem lies outside the GAL16v8?

**Reply by rehsd** [reply]
Yes, likely this is a user issue.  :)  I will proceed using the "clock input" component on pin 1 and troubleshoot the circuit outside of the GAL. My initial thought was that I was incorrectly using the clock input component, but I am taking your feedback as the clock input component on pin 1 should work fine for pulling an external clock into the GAL. Thank you for your quick response and help!
